### PR TITLE
03_zadanie

### PR DESCRIPTION
**Pull request by wsparcie**

### CODE REVIEW

**Review by Patmiko — APPROVED** [review]
Wszystko ok możnaby tylko troche bardziej pokazać na jakichś grafach przebieg treningu wraz ze zmianą hiperparametrów. Ogólnie dobrze rozwiązane zadanie. 
Powodzenia dalej w zadaniach (dziwne że nikt inny nie dał review XDD)

**Comment by sz3kz** [comment]
*Podobało* mi się **uzasadnienie wartosci hiperparametrów** `image_size`, `epochs` oraz `lr`. Porównując z kodem bazowym widziałem, że zmienione zostały także wartości `num_workers`(2->0) oraz `batch_size`(8->16), *z chęcią* przeczytałbym dlaczego i te parametry zmieniłeś.

Ciekawa dla mnie była implementacja **augmentacji danych** w klasie `VOCResizeWrapper`, tym bardziej że został wprowadzony **nietrywialny patch** w postaci flagi `is_train`, za pomocą której można użyć klasy zarówna dla **Training Set**  jaki i **Validation Set**, nie wprowadzając augmentacji danych do Validation Setu(co można łatwo przeoczyć).

*Najciekawsze* dla mnie było **obniżenie kary za pomyłki na tle**, z czym mój model ma najwięcej problemu.
Wydaje mi się, że to jest powodem iż według metryk `train_loss` oraz `val_pixel_acc` jest troche gorsze niż np moje. Co mnie najbardziej zaskoczyło to to iż chociaż metryki są gorsze, to wizualizacja pokazuje o wiele lepsze maski.

*Dodatkowy plus* za to, że pokazana została **wizualizacja** jak model radzi sobie zarówno na **Test Set** jaki i **Validation Set**.

Schludny kod, komentarze dodane tam gdzie potrzeba.

**Review by jakubbaldyga — APPROVED** [review]
Hej, myślę że jest git. Jest przejrzyście i wszystko wyjaśnione, fajnie że uwzględniłeś działanie optymalizatora adam i pomniejszyłeś wartość tła.
